src/session/content/chat_history.rs

```diff
@@ -223,4 +228,148 @@ impl ChatHistory {
         let adj = imp.list_view.vadjustment().unwrap();
         self.load_older_messages(&adj);
     }
+
+    fn update_last_viewed_message(&self, chat: &Chat) {
+        // The timeout seems to be necessary. because the `ListView` lags behind when the
+        // vadjustment has changed. In consequence, some visible messages could not be marked as
+        // read.
+        // A positive effect of the timeout is that the search for the bottommost message won't be
+        // triggered immediately after the vadjustment value has changed. This is an advantage when
+        // someone scrolls through the history very fast. The search for the bottommost message will
+        // begin only after scrolling has stopped.
+        let source_id = glib::timeout_add_local_once(
+            std::time::Duration::from_millis(200),
+            clone!(@weak self as obj, @weak chat => move || {
+                let imp = obj.imp();
+
+                imp.view_messages_source_id.take();
+
+                let list =  imp.list_view.observe_children();
```

Comment:
I don't think this is the right approach. This function is [heavily discouraged by gtk docs](https://docs.gtk.org/gtk4/method.Widget.observe_children.html) because, other than returning the list of children, it also tracks them using the listmodel signals, which slowdowns things. I think we should use first_child() and next_sibling() functions to iterate children, like we do [here](https://github.com/melix99/telegrand/blob/6c3372207d16bb0a3d5853751cfca12bedd92b9c/src/session/content/message_row/base.rs#L71).

Reply:
Btw, I've checked and observe_children() also iterates using next_sibling(), but it does that using a custom listmodel + tracks children, so it does extra overhead which we don't need anyway.

Reply:
This is indeed bad. And binary search is also counterproductive as this is a linked list without random access. So I think just iterating the children sequentially is the best thing we can do.